packages/wrangler/scripts/tsc-all.ts

```diff
@@ -0,0 +1,88 @@
+// Runs `tsc` against all `tsconfig.json`s in the current working directory.
+// This ensures all files are type-checked with the correct configuration.
+
+import childProcess from "node:child_process";
+import events from "node:events";
+import fs from "node:fs";
+import path from "node:path";
+import readline from "node:readline";
+import chalk from "chalk";
+
+const colours = [
+	chalk.red,
+	chalk.yellow,
+	chalk.green,
+	chalk.cyan,
+	chalk.blue,
+	chalk.magenta,
+];
+
+const argv = process.argv.slice(2);
+
+function* walkTsConfigs(root: string): Generator<string> {
+	const entries = fs.readdirSync(root, { withFileTypes: true });
+	for (const entry of entries) {
+		if (entry.name === "node_modules") continue; // Ignore `node_modules`s
+		const entryPath = path.join(root, entry.name);
+		if (entry.isDirectory()) {
+			yield* walkTsConfigs(entryPath);
+		} else if (entry.name === "tsconfig.json") {
+			yield entryPath;
+		}
+	}
+}
+
+// Find all `tsconfig.json`'s in the current directory
+const tsConfigs = Array.from(walkTsConfigs(process.cwd())).sort();
+
+const prefixes = tsConfigs.map((tsConfig) =>
+	path.dirname(path.relative("", tsConfig))
+);
+const maxPrefixLength = Math.max(...prefixes.map((prefix) => prefix.length));
+
+type ExitEventArguments = [code: number | null, signal: NodeJS.Signals | null];
+function runTsc(i: number, watch = false): Promise<ExitEventArguments> {
+	const tsConfig = tsConfigs[i];
```

Comment:
Can you pass in `tsConfig`, `prefix`, and `colour` instead of reaching out to the global scope?